File: app/src/main/java/pl/piotrskiba/angularowo/models/Ban.java
package pl.piotrskiba.angularowo.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Ban implements Serializable {

    public final static String TYPE_BAN = "Ban";
    public final static String TYPE_MUTE = "Wyciszenie";
    public final static String TYPE_WARNING = "Ostrzeżenie";

    @SerializedName("name")
    private final String username;
    private final String reason;
    private final String banner;
    @SerializedName("bantime")
    private final float banTime;
    @SerializedName("expires")
    private final float expireDate;
    private final String type;

    public Ban(String username, String reason, String banner, float banTime, float expireDate, String type){
        this.username = username;
        this.reason = reason;
        this.banner = banner;
        this.banTime = banTime;
        this.expireDate = expireDate;
        this.type = type;
    }

    public String getUsername() {
        return username;
    }

    public String getReason() {
        return reason;
    }

    public String getBanner() {
        return banner;
    }

    public float getBanTime() {
        return banTime;
    }

    public float getExpireDate() {
        return expireDate;
    }

    public String getType() {
        return type;
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/models/Ban.kt
package pl.piotrskiba.angularowo.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Ban(
        @field:SerializedName("name")
        val username: String,
        val uuid: String,
        val reason: String,
        val banner: String,
        @field:SerializedName("bantime")
        val banTime: Float,
        @field:SerializedName("expires")
        val expireDate: Float,
        val type: String) : Serializable {

    companion object {
        const val TYPE_BAN = "Ban"
        const val TYPE_MUTE = "Wyciszenie"
        const val TYPE_WARNING = "Ostrzeżenie"
        const val TYPE_KICK = "Wyrzucenie z gry"
    }

}